mlflow/store/tracking/databricks_rest_store.py

```diff
@@ -91,38 +92,37 @@ def start_trace(self, trace_info: TraceInfo) -> TraceInfo:
             The returned TraceInfo object from the backend.
         """
         try:
-            sql_warehouse_id = MLFLOW_TRACING_SQL_WAREHOUSE_ID.get()
-            req_body = message_to_json(
-                CreateTrace(
-                    trace_info=trace_info_to_proto(trace_info), sql_warehouse_id=sql_warehouse_id
-                )
-            )
-            if uc_schema := trace_info.trace_location.uc_schema:
-                location = f"{uc_schema.catalog_name}.{uc_schema.schema_name}"
-            # TODO: we should check if the experiment has a span location tag
-            elif mlflow_experiment := trace_info.trace_location.mlflow_experiment:
-                location = mlflow_experiment.experiment_id
-            else:
-                raise MlflowException("Invalid trace location")
-            response_proto = self._call_endpoint(
-                CreateTrace,
-                req_body,
-                endpoint=f"{_V4_REST_API_PATH_PREFIX}/mlflow/traces/{location}",
-                retry_timeout_seconds=MLFLOW_ASYNC_TRACE_LOGGING_RETRY_TIMEOUT.get(),
-            )
-            return TraceInfo.from_proto(response_proto.trace_info)
+            if trace_info.trace_location.uc_schema is not None:
+                return self._start_trace_v4(trace_info)
+
         # Temporarily we capture all exceptions and fallback to v3 if the trace location is not uc
         # TODO: remove this once the endpoint is fully rolled out
         except Exception as e:
-            if isinstance(e, MlflowException) and e.error_code == ErrorCode.Name(
-                ENDPOINT_NOT_FOUND
-            ):
-                _logger.debug("Server does not support CreateTrace API yet.")
-            if trace_info.trace_location.mlflow_experiment is not None:
-                _logger.debug("Falling back to V3 API.")
-                return super().start_trace(trace_info)
-            else:
+            if trace_info.trace_location.mlflow_experiment is None:
+                _logger.debug("MLflow experiment is not set for trace, cannot fallback to V3 API.")
                 raise
+            _logger.debug(f"Falling back to V3 API due to {e!s}")
+        return super().start_trace(trace_info)
+
+    def _start_trace_v4(self, trace_info: TraceInfo) -> TraceInfo:
+        location, otel_trace_id = parse_trace_id_v4(trace_info.trace_id)
+        if location is None:
+            raise MlflowException("Invalid trace ID format for v4 API.")
+
+        req_body = message_to_json(
+            CreateTraceInfo(
+                location_id=location,
+                trace_info=trace_info_to_proto(trace_info),
+            )
```

Comment:
@B-Step62 isn't the req_body for this supposed to be just the `TraceInfo` converted to json?